### DISCUSSION TITLE
still dont understand :/

### DISCUSSION THREAD

**Discussion by MAngel666** [mpounsett/nagiosplugin#52] [Q&A]
Hi

after multiple reading of the tutorial I still dont understand the logic :/

Am I on the right way? :)

```
import nagiosplugin

class Check_A(nagiosplugin.Resource):
    def probe(self):
        a = 1
        return [nagiosplugin.Metric('a', a, context='a')]

class Check_A(nagiosplugin.Context):
    def evaluate(self):
        if a == 1:
            return nagiosplugin.Result(nagiosplugin.Ok, 'a is 1', context='a')
        if a == 2:
            return nagiosplugin.Result(nagiosplugin.Critical, 'a is 2', context='a')

@nagiosplugin.guarded
def main():
    check = nagiosplugin.Check(
            Check_A(),
            nagiosplugin.Context('a')
    )

    check.main()

if __name__ == '__main__':
    main()
```

**Comment by mpounsett** [comment]
Sorry, it looks like nobody picked up this question.

Questions about the library aside, you've defined two classes with the same name, which won't work in Python.  The second definition will overwrite the first one, so you've got no way to refer to your Resource class.

As for the nagios library usage, I'm not sure what you're trying to accomplish.  You wouldn't normally be checking against static integers like that.... the idea behind a plugin is that you should be able to set ranges for values to be in, compare those ranges to a metric which you obtain, and return a result after comparing the two.   If all you want to do is return results based on static values you don't need nagiosplugin .. you could just use an `if` statement and `exit` with the right values to return a result to your monitoring system.

Have you looked at the tutorials?  The [check_load](https://nagiosplugin.readthedocs.io/en/stable/tutorial/check_load.html) example is about as simple as it gets, and you should be able to adapt that to just about any simple case.   If that doesn't seem to fit with your use case, then it would help if you explained what it is you're trying to write a check for, and what's different about it that makes the structure of the examples not work for you.
